File: auth-service/src/main/java/com/shopflow/auth/service/AuthService.java
package com.shopflow.auth.service;

import com.shopflow.auth.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface AuthService {

    InternalUserResponse getUserById(UUID userId);

    void register(RegisterRequest request);

    void verifyEmail(VerifyOtpRequest request);

    void resendOtp(String email);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(String refreshToken);

    void logout(String accessToken, String email);

    void forgotPassword(String email);

    void resetPassword(String email, String otp, String newPassword);

    Page<InternalUserResponse> getAllUsers(Pageable pageable);
    void banUser(UUID userId);
    void unbanUser(UUID userId);
}

File: shopflow-frontend/src/services/modules/auth.service.js
import api from '../api/axiosInstance';

export const authService = {
  login: async (credentials) => {
    const response = await api.post('/auth/login', credentials);
    if (response.data.access_token) {
      localStorage.setItem('token', response.data.access_token);
      localStorage.setItem('refreshToken', response.data.refresh_token);
    }
    return response.data;
  },
  
  register: async (userData) => {
    return await api.post('/auth/register', userData);
  },
  
  verifyEmail: async (data) => {
    return await api.post('/auth/verify-email', data);
  },
  
  resendOtp: async (email) => {
    return await api.post('/auth/resend-otp', { email });
  },
  
  forgotPassword: async (email) => {
    return await api.post('/auth/forgot-password', { email });
  },
  
  resetPassword: async (data) => {
    return await api.post('/auth/reset-password', data);
  },
  
  logout: async () => {
    try {
      await api.post('/auth/logout');
    } finally {
      localStorage.removeItem('token');
      localStorage.removeItem('refreshToken');
    }
  }
};
